models/agent_helpers.go

```diff
@@ -447,6 +447,10 @@ func RemoveAgent(q *reform.Querier, id string, mode RemoveMode) (*Agent, error)
 		return nil, err
 	}
 
+	if id == PMMServerAgentID {
+		return nil, status.Error(codes.PermissionDenied, "Removing pmm-agent on PMM Server prevents remote monitoring.")
```

Comment:
But does not this agent monitor **local** PostgreSQL by default as well?
If yes, maybe we need a more general error message here.

Reply:
Yes, it is, but it's not critical.